python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py

```diff
@@ -52,19 +59,32 @@ def time_window_partition_scope(self) -> Optional[datetime.timedelta]:
     @public
     @staticmethod
     def eager() -> "AutoMaterializePolicy":
-        return AutoMaterializePolicy(
-            on_missing=True,
-            on_new_parent_data=True,
-            for_freshness=True,
-            time_window_partition_scope_minutes=datetime.timedelta.resolution.total_seconds() / 60,
-        )
+        return _EAGER_POLICY
 
     @public
     @staticmethod
     def lazy() -> "AutoMaterializePolicy":
-        return AutoMaterializePolicy(
-            on_missing=True,
-            on_new_parent_data=False,
-            for_freshness=True,
-            time_window_partition_scope_minutes=datetime.timedelta.resolution.total_seconds() / 60,
-        )
+        return _LAZY_POLICY
+
+    @property
+    def policy_type(self) -> AutoMaterializePolicyType:
```

Comment:
I think I'd prefer if, for now, we just made this a function of `self.on_new_parent_data`. I.e. if that's true, you're eager, if it's false, you're lazy. We can add some code in the __new__ method on this class to enforce that you never have `on_new_parent_data` and `for_freshness` as both false (meaning `not on_new_parent_data` would imply `for_freshness`, which would mean that you're lazy)

Reply:
that seems reasonable... how would you make the distinction once we have on_new_code_version?

Reply:
Gotcha, so lazy = `for_freshness` and not (`on_new_parent_data` or `on_new_code_version`). eager = `on_new_parent_data` or `on_new_code_version`, `for_freshness` can be on or off